### PR TITLE
ping: Allow to ping IPv6 link-local scope address without link specification

### LINKED ISSUE #99: Broken ping to local interfaces with IPv6
This bug was introduced in #57.

    $ ip addr |grep 'inet6'
        inet6 fe80::fc54:ff:fe2d:cc21/64 scope link
    
    $ ./ping -6 fe80::fc54:ff:fe2d:cc21
    connect: Invalid argument

### PR DESCRIPTION
**Pull request by pevik**
This reverts commit e25568f5e580d5631f533ae8474bcca40a011690.

as it breaks ping to local IPv6 interfaces:

$ ip addr |grep 'inet6'
    inet6 fe80::fc54:ff:fe2d:cc21/64 scope link

$ ./ping -6 fe80::fc54:ff:fe2d:cc21
connect: Invalid argument

Fixes #99 (regression in #57).

### CODE REVIEW

**Comment by pevik** [comment]
@jsynacek can you please review it?

**Comment by pevik** [comment]
@jsynacek OK, correct exit code is important. I created proper fix (although I'm not sure if using `IN6_IS_ADDR_MC_LINKLOCAL` does make sense).

I also agree that ping code is hard to follow (it would be nice to have some tests for it).

**Comment by okias** [comment]
offtopic: Totally agree with @jsynacek let's hope that Rust ping port [1] move forward soon :) 

[1] https://github.com/mikeecb/ring

**Comment by pevik** [comment]
@okias @jsynacek : Could you please review new version?

**Comment by pevik** [comment]
@jsynacek Sure, I tested it, works as expected.

**Comment by okias** [comment]
Thank you guys! :)
